### DISCUSSION TITLE
Document Function.length

### DISCUSSION THREAD

**Discussion by Fancy2209** [airsdk/Adobe-Runtime-Support#3488] [Ideas]
Just thought it would be nice if this was documented since it exists.
![imagem](https://github.com/user-attachments/assets/9bbdb395-33a0-4e8d-891b-0ee911f8a04c)

**Comment by ajwfrost** [comment]
Sounds like a good idea ... fyi, it's the number of arguments that the function expects. Will get this in for the next AS3 documentation update we do.

**Comment by shaucky** [comment]
How about `MovieClip.addFrameScript()`? It's also an undisclosed function/member.

**Reply by ajwfrost** [reply]
The internal comments suggest that this function can cause problems if you call it multiple times; it's intended for tools like Animate to use when generating a SWF file. If you then call it manually, you could cause instabilities (apparently!).

Probably a better option might be to make it more robust and turn it into a supported feature...! we can look into this...

thanks

**Reply by bobaoapae** [reply]
WoW, this fuction marked as "could cause instabilities" make me very sad. This it's used a lot on my application. Just a kick search on main project, without search on others swf files already compiled.

<img width="596" height="378" alt="image" src="https://github.com/user-attachments/assets/0cc585ef-905c-4b54-9d81-c050d68b2b5a" />

**Reply by ajwfrost** [reply]
Looking at the code, I think the warning is mostly because you have to use the method properly: if you don't, it could cause some odd behaviours, and there's no real checks or validation on the inputs. So e.g. it needs to be called with an even number of parameters, the first of each pair is the frame number integer (zero-based) and the second is a script/method to call. It looks to me as if calling it repeatedly would mean you overwrite an earlier entry but there may be GC issues if you do that from within an anonymous script that's the one that would be replaced!

But for normal/sensible usage I can't see anything that makes it look especially risky..
